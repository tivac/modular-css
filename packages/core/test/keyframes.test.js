"use strict";

var dedent = require("dedent"),
    namer  = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("scoping", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                namer
            });
        });

        it("should leave unknown animation names alone", () =>
            processor.string(
                "./unknown-name.css",
                dedent(`
                    .a { animation: a; }
                    .b { animation-name: b; }
                `)
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );
        
        it("should update scoped animations from the scoping plugin's message", () =>
            processor.string(
                "./animation.css",
                dedent(`
                    @keyframes a {}
                    .b { animation: a; }
                `)
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );

        it("should update the animation-name property", () =>
            processor.string(
                "./animation-name.css",
                dedent(`
                    @keyframes a {}
                    .b { animation-name: a; }
                `)
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );

        // Issue 208
        it("should update multiple animations properly", () =>
            processor.string(
                "./multiple-animations.css",
                dedent(`
                    @keyframes a {}
                    @keyframes b {}
                    .c { animation: a 10s linear, b 0.2s infinite; }
                `)
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );

        it("should update scoped prefixed animations from the scoping plugin's message", () =>
            processor.string(
                "./prefixed-animations.css",
                dedent(`
                    @-webkit-keyframes a {}
                    .b { animation: a; }
                `)
            )
            .then(() => processor.output())
            .then((result) =>
                expect(result.css).toMatchSnapshot()
            )
        );
    });
});
