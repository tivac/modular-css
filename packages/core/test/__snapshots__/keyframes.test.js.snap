// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js scoping should leave unknown animation names alone 1`] = `
"/* unknown-name.css */
.a {
    animation: a
}
.b {
    animation-name: b
}"
`;

exports[`/processor.js scoping should update multiple animations properly 1`] = `
"/* multiple-animations.css */
@keyframes a {}
@keyframes b {}
.c {
    animation: a 10s linear, b 0.2s infinite
}"
`;

exports[`/processor.js scoping should update scoped animations from the scoping plugin's message 1`] = `
"/* animation.css */
@keyframes a {}
.b {
    animation: a
}"
`;

exports[`/processor.js scoping should update scoped prefixed animations from the scoping plugin's message 1`] = `
"/* prefixed-animations.css */
@-webkit-keyframes a {}
.b {
    animation: a
}"
`;

exports[`/processor.js scoping should update the animation-name property 1`] = `
"/* animation-name.css */
@keyframes a {}
.b {
    animation-name: a
}"
`;
